genienlp/predict.py

```diff
@@ -218,11 +220,18 @@ def run(args, device):
             else:
                 confidence_estimators = None
             with torch.cuda.amp.autocast(enabled=args.mixed_precision):
-                generation_output = generate_with_model(model, it, model.numericalizer, task, args,
-                                                     original_order=original_order,
-                                                     output_confidence_features=args.save_confidence_features,
-                                                     confidence_estimators=confidence_estimators,
-                                                     disable_progbar=False)
+                if isinstance(model, TransformerForTokenClassification):
+                    generation_output = generate_with_classification_model(model, it, model.numericalizer, task,
```

Comment:
It would be cleaner if `generate_with_classification_model()` and `generate_with_seq2seq_model()` are combined to prevent if statements everywhere. What do you think?

Reply:
The underlying function to get outputs are different for each model. `generate_with_seq2seq_model` uses `generate()` whereas `generate_with_classification_model` uses `forward()`. Also `cross_attentions` or `confidence` are not supported for token classification. I can move the if-else logic to a function by passing `is_classification` as an argument and deciding there which `generate_with` to use.